pkg/controller/utils.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/deis/builder/pkg/conf"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	hostEnvName = "DEIS_WORKFLOW_SERVICE_HOST"
+	portEnvName = "DEIS_WORKFLOW_SERVICE_PORT"
+)
+
+// UserInfo represent the required information from a user to make a push and interact with deis/workflow
+type UserInfo struct {
+	Username    string
+	Key         string
+	FingerPrint string
```

Comment:
This should probably be `Fingerprint` without InterCaps (like `Username`). Not worth holding up this PR though.